src/base/DiagnosticPositionInfo.zig

```diff
@@ -0,0 +1,178 @@
+//! This module provides helpers for calculating position for diagnostics
+//! including the start and end line and column information
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+
+start_line: u32,
+start_col: u32,
+end_line: u32,
+end_col: u32,
+line_text: []const u8,
+
+const DiagnosticPositionInfo = @This();
+
+/// Finds the line number for a given position in the source
+fn lineNum(newlines: std.ArrayList(u32), pos: u32) u32 {
+    if (newlines.items.len == 0) {
+        return 0;
+    }
+
+    var lineno: u32 = 0;
+
+    while (lineno + 1 < newlines.items.len) {
+        if (newlines.items[lineno + 1] > pos) {
+            return lineno;
+        }
+        lineno += 1;
+    }
+
+    return lineno;
+}
+
+/// Gets the column number for a position on a given line
+fn columnNum(newlines: std.ArrayList(u32), line: u32, pos: u32) !u32 {
```

Comment:
I don't think this is correct. Though I guess it depends on if newlines always starts with a `0` entry and how you count.

If I have newlines = `[10, 20, 30]` and a positions of `5`.
The line is `0`.

This code would then load `10` as the line start, but in reality, the line start is `0`.
So I think everything is offset here.